packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -870,7 +871,8 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
       // back gesture occurs but no predictive back route transition exists to
       // handle it. The back gesture should still cause normal pop even if it
       // doesn't cause a predictive transition.
-      return handlePopRoute();
+      await handlePopRoute();
```

Comment:
is there a lint that suggests this? seems a bit weird...

Reply:
yes _handleCommitBackGesture is a  Future<void>  and handlePopRoute is a Future<bool> here

Reply:
ah I see. looks good then